src/metabase/api/dashboard.clj

```diff
@@ -800,21 +800,23 @@
              :when field-id]
          field-id)))
 
-(defn- param-key->field-ids
+(defn- param-key->param
   "Get Field ID(s) associated with a parameter in a Dashboard.
 
-    (param-key->field-ids (t2/select-one Dashboard :id 62) \"ee876336\")
+    (param-key->param (t2/select-one Dashboard :id 62) \"ee876336\")
     ;; -> #{276}"
   [dashboard param-key]
   {:pre [(string? param-key)]}
-  (let [{:keys [resolved-params]} (t2/hydrate dashboard :resolved-params)
-        param                     (get resolved-params param-key)]
-    (mappings->field-ids (:mappings param))))
+  (let [{:keys [resolved-params]} (t2/hydrate dashboard :resolved-params)]
+    (get resolved-params param-key)))
 
 (defn- chain-filter-constraints [dashboard constraint-param-key->value]
   (into {} (for [[param-key value] constraint-param-key->value
-                 field-id          (param-key->field-ids dashboard param-key)]
-             [field-id value])))
+                 :let              [param (param-key->param dashboard param-key)]
+                 field-id          (mappings->field-ids (:mappings param))]
+             (case (:type param)
+               :string/!= [field-id [:!= value]]
```

Comment:
Yeah. The problem is that I'm still not sure that's the best place to handle this stuff... Plus all those `:string/starts-with` do not seem to generate correct SQL when used in a card, so I'm wondering if there is larger-scale problem.

I should've written those thoughts directly in the comment, I guess, rather than ponder on them myself. :-)